Wasted/Pages/Place/DealEditorPage.xaml.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rg.Plugins.Popup.Services;
+using Wasted.Utils;
+using Wasted.Utils.Exceptions;
+using Wasted.Utils.Services;
+using Wasted.WastedAPI;
+using Wasted.WastedAPI.Business_Objects;
+using Xamarin.Forms;
+using Xamarin.Forms.PlatformConfiguration;
+using Xamarin.Forms.PlatformConfiguration.iOSSpecific;
+using Xamarin.Forms.Xaml;
+using Entry = Xamarin.Forms.Entry;
+
+namespace Wasted.Pages.Place
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class DealEditorPage : ContentPage
+    {
+        public Deal SelectedDeal { get; set; }
+        public EntryLengthValidator Validator { get; set; }
+        
+        private string NewTitle => TitleEntry.Text;
+        private string NewCurrentCost => CurrentCostEntry.Text;
+        private string NewRegularCost => RegularCostEntry.Text;
+        private string NewDueDate => DueDatePicker.Date.ToString("yyyy-MM-dd");
+        private string NewDescription => DescriptionEntry.Text;
+
+        private CurrentUserService _service = DependencyService.Get<CurrentUserService>();
+        
+        public DealEditorPage(Deal selectedDeal)
+        {
+            InitializeComponent();
+            On<iOS>().SetUseSafeArea(true);
+            
+            Validator = new EntryLengthValidator(maxEntryLength: 50);
+            SelectedDeal = selectedDeal;
+            BindingContext = this;
+        }
+
+        /// <summary>
+        /// Checks whether all user fields are not empty
+        /// </summary>
+        /// <returns>'false' if any of fields is empty. 'true' if all fields are not empty</returns>
+        private bool IsDataValid()
+        {
+            return !StringUtils.AnyNullOrEmpty(NewTitle, NewCurrentCost, 
```

Comment:
We already have a similar function for checking if data is empty I think in one of exception utils classes